### DISCUSSION TITLE
Change "Attach Exisiting" to display related record field (other than id)

### DISCUSSION THREAD

**Discussion by titantwentyone** [filamentphp/filament#628] [Help] [answered]
I can see that a `BelongsToManySelect` field is coming soon and would solve my issue below as I could tackle this a different way. It is already available in Forms and will (I assume!) be available in Admin later on this year. I just wondered if there was a way to mimic this functionality using the existing tools we have.

My set up is that I have a Many To Many relationship where many Customers can be booked onto many Events. Simplified table structure is as follows:
```
customer - id, name
event - id, date
booking - id, customer_id, event_id
```
I want my users to be able to add bookings manually on the Event page. In lieu of the `BelongsToManySelect` I can make using of the "Attach Exisiting" functionality for my relationship as an attachment can only be to customers and events that already exist in the system. This works but only displays the `id` of a related record (i.e. customer) and I would ideally like to provide the customers "name" field so its easier to read. Is there a way to override this so that another field is displayed but the id is passed back for the relationship?

![image](https://user-images.githubusercontent.com/1583366/135762532-adee36d9-7144-4ccb-8019-3190adcd826f.png)

**Comment by danharrin** [comment, marked as answer]
https://filamentadmin.com/docs/1.x/admin/resources#managing-multiple-related-records

Does setting this work?

```php
public static $primaryColumn = 'name';
```

**Reply by titantwentyone** [reply]
Thanks Dan. I think in my various iterations of trying to mimic a `BelongsToManySelect` I missed this!
